packages/nightingale-saver/src/nightingale-saver.ts

```diff
@@ -49,11 +52,17 @@ class NightingaleSaver extends NightingaleElement {
     if (typeof this.preSave === "function") {
       this.preSave();
     }
-    const { width, height } = element.getBoundingClientRect();
+    if (this.#height < 0 && this.#width < 0) {
```

Comment:
I think this doesn't consider the case where you have taken a snapshot, and then the element's size changes (e.g. when the tracks are responsive, and the window has been changed).

What if instead of saving previous dimensions, we make sure the `canvas.preview` is removed before getting the size?